### PR TITLE
Rename tree-sitter-c fork to avoid confusion

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Rename our C grammar into `rizin-grammar-c`

**Test plan**

CI is green

### CODE REVIEW

File: subprojects/tree-sitter-c.wrap [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [wrap-git]
 url = https://github.com/rizinorg/tree-sitter-c
```
Comment (XVilka) [inline subprojects/tree-sitter-c.wrap]: URL should be changed before merging - rename the repository first, then do the change, then try again.

File: meson.build [hunk 2]
```diff
@@ -300,15 +300,15 @@ if not tree_sitter_dep.found()
...
-# handle tree-sitter-c
-r = run_command(py3_exe, check_meson_subproject_py, 'tree-sitter-c', check: false)
+# handle rizin-grammar-c
+r = run_command(py3_exe, check_meson_subproject_py, 'rizin-grammar-c', check: false)
 if r.returncode() == 1 and get_option('subprojects_check')
   error(subproject_clean_error_msg)
 endif
 
-tree_sitter_c_dep = dependency('tree-sitter-c', required: get_option('use_sys_tree_sitter'), static: is_static_build, fallback: [])
+tree_sitter_c_dep = dependency('rizin-grammar-c', required: get_option('use_sys_tree_sitter'), static: is_static_build, fallback: [])
```
Comment (wargio) [inline meson.build]: just rename it as `rz-grammar` instead of `rizin-grammar-c`

Comment (XVilka) [reply meson.build]: Too ambiguous, IMHO.

Comment (wargio) [reply meson.build]: ok, but also the binary called has been renamed.

**Review by wargio — APPROVED** [review]

**Review by Rot127 — APPROVED** [review]

**Pull request merged**